features/FEATURE_LWIP/lwip-interface/lwip_stack.c

```diff
@@ -992,6 +1023,10 @@ static nsapi_error_t mbed_lwip_socket_bind(nsapi_stack_t *stack, nsapi_socket_t
         return NSAPI_ERROR_PARAMETER;
     }
 
+    if (!ip_addr_isany(&ip_addr) && !mbed_lwip_is_local_addr(&ip_addr)) {
+        return NSAPI_ERROR_PARAMETER;
```

Comment:
I wonder if the return error here should be "NO_ADDRESS"? Guess this is fine though.

Reply:
I see Nanostack has "PARAMETER" already, so this matches.